lib/chef/knife/ri_list.rb

```diff
@@ -0,0 +1,137 @@
+#
+# Author:: Jimmy Coppens (jimmy@yhmg.com)
+# Copyright:: Copyright (c) 2010-2011 Opscode, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require 'chef/knife/ec2_base'
+
+class Chef
+    class Knife
+        class RiList < Knife
+            
+            include Knife::Ec2Base
+            
+            banner "knife ri list - List all reserved instance leases defined in the aws cloud"
+            
+            def build_ec2Map
+                ec2Servers = connection.servers
+                
+                hash_map = ec2Servers.inject({}) do |hash, o|
+                  if o.state == "running"
+                      if o.vpc_id
+                          vpc_or_not = " vpc"
+                      else
+                          vpc_or_not = " ec2"
+                      end
+                      
+                      key = o.flavor_id.to_s + " " + o.availability_zone.to_s + vpc_or_not
+                      hash[key] ||= 0
+                      hash[key] += 1
+                  end
+
+                  hash
+
+                end
+                
+                hash_map
+            
+            end
+
+            def build_riMap(ri_leases)
+                
+                hash_map = ri_leases.inject({}) do |hash, o|
+                    if o["reservedInstancesId"]
```

Comment:
I think we should rename the variable 'o' to ri_detail variable, it will improve the readability of the code